backend/src/ConnectionPool.ts

```diff
@@ -86,9 +86,15 @@ export default class ConnectionPool {
   set = (connection: IConnection, setCLI?: boolean) => {
     if (!connection) Logger.warn('No connections to set!', { connection })
     let instance = this.find(connection.id)
-    if (instance) instance.set(connection, setCLI)
-    else instance = this.add(connection)
-    this.updated()
+    if (instance) {
+      if (JSON.stringify(connection) !== JSON.stringify(instance.params)) {
```

Comment:
could you validate that instance is a object?

Reply:
In this case it doesn't really matter if it's an object or not - just if we've found it in the collection.